Makefile

```diff
@@ -0,0 +1,28 @@
+SOURCE_FILES?=$$(glide novendor)
+TEST_PATTERN?=.
+TEST_OPTIONS?=
+
+setup: ## Install all the build and lint dependencies
+	curl https://glide.sh/get | sh
+	go get -u github.com/kisielk/errcheck
+	go get -u github.com/golang/lint/golint
+	glide install
+
+test: ## Run all the tests
+	go test $(TEST_OPTIONS) -cover $(SOURCE_FILES) -run $(TEST_PATTERN) -timeout=30s
+
+lint: ## Run all the linters
+	errcheck $(SOURCE_FILES)
+	go vet $(SOURCE_FILES)
+	golint ./... | grep -v vendor
```

Comment:
this will hide the exit code of `golint`.

I think it should be something like 

```sh
find . -type d -depth 1 -not -wholename './vendor/*' | while read -r d; do
  golint $d
done
```